src/components/drawer/drawer.component.js

```diff
@@ -21,20 +22,63 @@ const Drawer = ({
   animationDuration,
   ...props
 }) => {
+  const drawerSidebarContentRef = useRef();
   const isControlled = expanded !== undefined;
   const [isExpandedInternal, setIsExpandedInternal] = useState(defaultExpanded || false);
-  const [isFirstOpen, setFirstOpen] = useState(true);
+  const [isOpening, setIsOpening] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
   const isExpanded = isControlled ? expanded : isExpandedInternal;
+
+  const duration = useCallback(() => {
```

Comment:
this is a method so it's name should be a verb to increase readability, e.g.: `getDuration`or `sanitizeDuration`

Reply:
renamed to `getAnimationDuration`